### ISSUE TITLE
1.0.10 RF Page system malfunctions

### ISSUE THREAD

**Issue by jfint** [FrSkyRC/ETHOS-Feedback-Community#782]
Tandem X20
Ethos 1.0.10 release
TD-ISRM 2.1.9
Archer GR8 2.1.8, registered at UID 0
Archer RS 2.1.8, registered at UID 1

I'm trying to use SBUS out from the Archer RS to the SBUS in of the GR8 to create a backup reciever, at first I thought it was working, then I tried verifying via logged the "RX" in the telemetry, and that led to the following video.  In it you will see

- RX switchover simply by visiting the info page in system
- reset Bind not working
- A receiver working with the TX, but not bound to the TX
- probably some other oddity.  
I think Ethos has some issue with its RF page

https://www.youtube.com/watch?v=GtwPsReGtuQ

I apologize for the vid quality

**Comment by jfint** [comment]
For contrast, this is the behavior I would have expected to happen, exhibited when there is just 1 receiver.

https://www.youtube.com/watch?v=dkRINztSAls

**Comment by pld74** [comment]
I noticed that same behaviour in 1.0.8 and I found if I waited for the reset command to be sent from the transmitter to the receiver it would reset it. Sometimes took about 10-15 seconds. Not sure if this helps.

**Comment by pld74** [comment]
I noticed that same behaviour in 1.0.8 and I found if I waited for the reset command to be sent from the transmitter to the receiver it would reset it. Sometimes took about 10-15 seconds. Not sure if this helps. I’ll have to test this in 1.0.10 and report back with dual receivers.

**Comment by pld74** [comment]
**Update:**

I can repeat the same peculiar reset bind behaviour as @jfint using R8 Pro and M+ receiver and still reporting telemetry and transmitter shows no receiver bind info. I also tried to power cycle receiver and transmitter, same peculiar behaviour.

Also, I can repeat the same peculiar behaviour happens when you select the Reset Receiver Option. The receiver does not reset registration.

Hardware Details:
X20S Ethos 1.0.10 ISRM FW 2.1.9 FCC
R8 Pro FW 2.1.8
M+ FW 2.1.8

**Comment by jfint** [comment]
Thank you for reproducing with more recievers.  As a result of this testing I am not at all confident that the backup receiver will work correctly.  Thr more I test the more i actually am confident it will not work correctly.  Which makes this a critical bug.  

Im not sure if its the same behavior with a 900mhz rx as the secondary.

**Comment by pld74** [comment]
@jfint This is from the Ethos 1.0.5 Manual, "maybe" we are misinterpreting Reset Bind Option:???

Reset Bind: If you change your mind about sharing a model, select 'Reset bind' to clean up and restore your bind. Power cycle the receiver, and it will be bound to your transmitter.

**Comment by jfint** [comment]
If that is the case then is is not working correctly in the second video I posted with the single receiver, and I'm open to that if that is the what is expected, even if I don't understand the logic behind it.  But I want consistent behavior for sure.

Also the first parts of the opening video is possibly more disturbing, where it changes receivers because I viewed the version numbers.

**Comment by pld74** [comment]
So I don't get that "change in receivers" but I am also not within 2 metres (6 feet) of the receivers. Too close cause RF Anomalies. Try it further away and maybe report back...

**Comment by jfint** [comment]
Does your reliably boot using RX 0 initially?

**Comment by pld74** [comment]
Yes BUT if I am within 12 inches on the antenna, 2/10 times it boots off of RX 1. That's because it is too close to the antenna. If I take one step away and place the TX there, 10/10 boots RX 0.

**Comment by jfint** [comment]
I just did it 6 ft away, with the exact same behavior.  I have to go to the info screen in order to get it to switch to RX 0, it starts on RX 1.

**Comment by pld74** [comment]
Did you register with same or unique UID?

**Comment by jfint** [comment]
I did the GR8 as UID 0, and the RS as UID1.  I do have telemetry on both on at the moment, and I agree that that might be odd, but having it on allowed me to gather this data:

I wrapped the nose of the glider in aluminum foil and attempted to get away from it far enough to switch to the RS, I DID get an appropriate switch, so that is fantastic, and I think that I am willing to fly again.  You can see that telemetry did indeed say that it was using RX 1 at the beginning, and I got it moved to RX 0 by going to the info screen.  It did indeed switch to RX 1 when the RF got crappy.

I would love to know if I am supposed to turn off the telemetry on RX 1, I feel like that is a bit FrSky shortcoming for that to be that case, it should at least be able to do the RSSi and VFR values for every receiver in the system.  The question becomes, can it do that if the second receiver is 900Mhz, because that alone is a reason to switch.

![image](https://user-images.githubusercontent.com/16928210/129416076-83b4da29-5d3c-4017-bcf1-1b33c934c516.png)

In summary that initial condition and the strange switch by going to the info screen, and the odd behavior of not resetting the bind the same way with 1 vs 2 receivers are the issues remaining on this bug I think.

**Comment by jfint** [comment]
@pld74  I'm confused on the UID thing, oh RCG it was posted that they should be different fro the multi-receiver trio functionality, but from your excerpt it seems I should have registered them on the same UID?

**Comment by pld74** [comment]
Yeah I read the same on RCG. Now I use the same UID unless as the Frsky manual says, to extend physical channels.

**Comment by pld74** [comment]
> I did the GR8 as UID 0, and the RS as UID1. I do have telemetry on both on at the moment, and I agree that that might be odd, but having it on allowed me to gather this data:
> 
> I wrapped the nose of the glider in aluminum foil and attempted to get away from it far enough to switch to the RS, I DID get an appropriate switch, so that is fantastic, and I think that I am willing to fly again. You can see that telemetry did indeed say that it was using RX 1 at the beginning, and I got it moved to RX 0 by going to the info screen. It did indeed switch to RX 1 when the RF got crappy.
> 
> I would love to know if I am supposed to turn off the telemetry on RX 1, I feel like that is a bit FrSky shortcoming for that to be that case, it should at least be able to do the RSSi and VFR values for every receiver in the system. The question becomes, can it do that if the second receiver is 900Mhz, because that alone is a reason to switch.
> 
> ![image](https://user-images.githubusercontent.com/16928210/129416076-83b4da29-5d3c-4017-bcf1-1b33c934c516.png)
> 
> In summary that initial condition and the strange switch by going to the info screen, and the odd behavior of not resetting the bind the same way with 1 vs 2 receivers are the issues remaining on this bug I think.

Been flying with no issues with firmware 2.1.8... GO FLY!!!

**Comment by pld74** [comment]
> I did the GR8 as UID 0, and the RS as UID1. I do have telemetry on both on at the moment, and I agree that that might be odd, but having it on allowed me to gather this data:
> 
> I wrapped the nose of the glider in aluminum foil and attempted to get away from it far enough to switch to the RS, I DID get an appropriate switch, so that is fantastic, and I think that I am willing to fly again. You can see that telemetry did indeed say that it was using RX 1 at the beginning, and I got it moved to RX 0 by going to the info screen. It did indeed switch to RX 1 when the RF got crappy.
> 
> I would love to know if I am supposed to turn off the telemetry on RX 1, I feel like that is a bit FrSky shortcoming for that to be that case, it should at least be able to do the RSSi and VFR values for every receiver in the system. The question becomes, can it do that if the second receiver is 900Mhz, because that alone is a reason to switch.
> 
> ![image](https://user-images.githubusercontent.com/16928210/129416076-83b4da29-5d3c-4017-bcf1-1b33c934c516.png)
> 
> In summary that initial condition and the strange switch by going to the info screen, and the odd behavior of not resetting the bind the same way with 1 vs 2 receivers are the issues remaining on this bug I think.



> I did the GR8 as UID 0, and the RS as UID1. I do have telemetry on both on at the moment, and I agree that that might be odd, but having it on allowed me to gather this data:
> 
> I wrapped the nose of the glider in aluminum foil and attempted to get away from it far enough to switch to the RS, I DID get an appropriate switch, so that is fantastic, and I think that I am willing to fly again. You can see that telemetry did indeed say that it was using RX 1 at the beginning, and I got it moved to RX 0 by going to the info screen. It did indeed switch to RX 1 when the RF got crappy.
> 
> I would love to know if I am supposed to turn off the telemetry on RX 1, I feel like that is a bit FrSky shortcoming for that to be that case, it should at least be able to do the RSSi and VFR values for every receiver in the system. The question becomes, can it do that if the second receiver is 900Mhz, because that alone is a reason to switch.
> 
> ![image](https://user-images.githubusercontent.com/16928210/129416076-83b4da29-5d3c-4017-bcf1-1b33c934c516.png)
> 
> In summary that initial condition and the strange switch by going to the info screen, and the odd behavior of not resetting the bind the same way with 1 vs 2 receivers are the issues remaining on this bug I think.

Did you try receiver reset? Don't think there is a Bind Reset now that we read the manual.

**Comment by Devil-67** [comment]
I have noticed the same thing for months with OpenTX, I use a GR8 and your RS as a backup RX.
Or also G-RX8 and RSXR.
I have informed Frsky directly about this, unfortunately as always no reaction.
You can find more info here in the forum: 
https://frsky-forum.de/index.php?thread/5194-g-rx8-und-als-redundanz-empf%C3%A4nger-rxsr/&pageNo=1

**Comment by pld74** [comment]
Let’s hope they see it here and resolve it or report back on our findings and they can guide us.

**Comment by pld74** [comment]
I have confirmed with FrkSy that in ACCESS with ARCHER receivers, that Telemetry Broadcasting can remain on for all receivers.

**Comment by jfint** [comment]
That's great, I hope that it is a step towards being able to deliver telemetry from both receivers at the same time, at least RSSI and VFR

**Comment by Devil-67** [comment]
ACCESS should have been able to do this a long time ago, at least RSSI and VFR should always be transmitted by both receivers.
For a redundant system this is a very important criterion to check if everything is working properly and if the antenna position is good.

**Comment by jfint** [comment]
@pld74 did you also get a definitive answer on if they are supposed to be the same UID or different?

**Comment by pld74** [comment]
Yes, Same UID. Different UID only if you’re splitting receivers to control physical channel output in groups of 8 hence the UID 0,1,2. Up to 24 servo outputs.

**Comment by pld74** [comment]
In summary to this thread's submitted issues:

1) UID: Use the same UID for Redundant Receivers. Use Unique UID only if you’re splitting receivers to control physical channel output in groups of 8 hence the UID Options of: 0, 1, 2. Up to 24 servo outputs can be controlled comprised of 3 receivers of 8 outputs each. If using Unique UID scenario, label each receiver with the UID number for reference as it’s important for proper servo output.

2) Reset Bind: If you change your mind about sharing a model, select 'Reset bind' to clean up and restore your bind. Power cycle the receiver, and it will be rebound to your transmitter.

2) Telemetry ON or OFF for ACCESS in Redundant Setup: Telemetry can remain ON for all receivers in a redundant setup. As a matte of fact, this is great for indepth logging in multiple receivers setups. You can use the log to better place the other receiver for optimal radio transmission reception.

4) Default Receiver at Power ON: Current behaviour is that even though a 'primary' receiver such as a GR8 or R8 Pro is used with a 'slave' receiver, the current protocol DOES NOT PRIORITIZE the primary receiver as the default transmission receiver during Power ON. I have communicated this to FrSky as an improvement that at Initial Power-Up the 'primary' receiver should ALWAYS connect first and switch to the redundant receiver based on RF Signal Metrics, as currently it does. I can understand how the current behaviour can be determined as a glitch but it is not, merely needs a refinement in the code which they are willing to refine.

Patience is a Virtue

Hope this helps everyone...

Cheers

**Comment by jfint** [comment]
The only discrepancy to what you say is reset bind(#2).  There are two functionalities, "Reset" and "Reset Bind".  "Reset Bind" has an issue with not behaving the same way every time.

**Reset**
This functionality only works if the RX is powered on and bound tot he TX, it will reset the connection between the receiver and the TX, and require the RX to be re-registered.  However this is only true of the RX is on that the time that it is selected, otherwise this functionality may be similar to "Reset Bind:

**Reset Bind**
This functionality has two behaviors as shown in the videos above:
** 1 RX** Reset bind breaks the RF link between the RX and the TX, however the RX can be re-bound without re-registering the receiver.  This is for sure the preferred behavior despite the description in the manual as it allows for predictable testing and installation, as well as a lack of unexpected/indeterminant behaviors.
**2 RXs** The RF link is not broken, and even after power cycle and there being no RX name displayed on the RF page the receiver is still bound and working, which is unexpected and makes testing an installation impossible.  This may be the behavior described in the manual, but the wording is very poor on the subject.

**Comment by jfint** [comment]
Thr UID information is also incomplete. 


https://www.rcgroups.com/forums/showpost.php?p=47654667&postcount=10404

**Comment by mawzthefinn** [comment]
I've left this open for a while, but it is a protocol limitation, not an ETHOS Issue.

I would suggest it be closed off for now, as these requests have been passed on to the RF team. ETHOS will support if the protocol behaviour changes

**Comment by mawzthefinn** [comment]
Closing off as per my previous post. This is RF level and has been passed on to the RF team.
